numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1755,36 +1773,66 @@ unpack_bits(PyObject *input, int axis)
         unpack_init = 1;
     }
 
-    NPY_BEGIN_THREADS_THRESHOLDED(PyArray_DIM(new, axis));
+    count = PyArray_DIM(new, axis) * 8;
+    if (outdims[axis] > count) {
+        in_n = count / 8;
+        in_tail = 0;
+        out_pad = outdims[axis] - count;
+    }
+    else {
+        in_n = outdims[axis] / 8;
+        in_tail = outdims[axis] % 8;
+        out_pad = 0;
+    }
 
-    n_in = PyArray_DIM(new, axis);
     in_stride = PyArray_STRIDE(new, axis);
     out_stride = PyArray_STRIDE(out, axis);
 
+    NPY_BEGIN_THREADS_THRESHOLDED(PyArray_Size(in) / PyArray_DIM(in, axis) * in_n);
```

Comment:
Can't you use PyArray_Size(out), maybe devide it by 8 if you like? `in` is breaking the tests.

About the other tests, there is a multiarray.py now which needs the change as well I think, but I did not quite follow that at the time, so am not quite sure without looking at it either.

Reply:
That is probably just a mistake -- the docstrings should have been moved, not copied. The redundant copy in _add_newdocs.py should be deleted.

Reply:
@shoyer Thanks for the clarification. @seberg Will fix momentarily, which should also make the previous "harmless" build issue go away. I was completely unaware of `multiarray.py` since I had grepped for `unpackbits` and opened the first likely occurrence (in `_add_newdocs.py`) without realizing there was duplication.

Reply:
Well, the last time around multiarray.py did not exist, so there is no way you could have known it.

Reply:
That makes sense. I am currently experiencing some network problems (internet works fine but git push/pull not so much). Will push what is hopefully a final version as soon as that gets resolved.